samza-sql/src/test/java/org/apache/samza/sql/avro/TestAvroRelConversion.java

```diff
@@ -231,6 +240,42 @@ public void testComplexRecordConversion() throws IOException {
     validateAvroSerializedData(serializedData, testStrValue);
   }
 
+  @Test
+  public void testComplexUnionConversionShouldWorkWithBothStringAndIntTypes() throws Exception {
```

Comment:
Were you able to repro the issue with this unit test by having the older changes (with cached schema) ?